src/components/AccountModule/utils.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react'
+import { useTheme } from '@aragon/ui'
+
+import { web3Providers } from '../../environment'
+import { getNetworkByChainId } from '../../network-config'
+import { getWeb3 } from '../../web3-utils'
+
+export const getWeb3Instances = () => {
+  const walletWeb3 = getWeb3(web3Providers.wallet)
+  const web3 = getWeb3(web3Providers.default)
+  return {
+    walletWeb3,
+    web3,
+  }
+}
+
+function normalizeNetworkName(chainId) {
+  return getNetworkByChainId(chainId).settings.shortName
+}
+
+export const useNetworkConnectionData = () => {
+  const [walletChainId, setWalletChainId] = useState()
+  const [clientChainId, setClientChainId] = useState()
+
+  useEffect(() => {
+    const { web3, walletWeb3 } = getWeb3Instances()
+    web3.eth.getChainId((err, chainId) => {
+      if (!err) {
+        setClientChainId(chainId)
+      }
+    })
+    walletWeb3.eth.getChainId((err, chainId) => {
```

Comment:
I'm wondering if we should do this call here, since it makes a network request. Maybe this is something we can bring up to where we poll for information, and encapsulate all of that in a context instead?

I believe we can also now subscribe to actual network change events from the provider spec.

Reply:
This is done in `useWallet()` now: https://github.com/aragon/aragon/pull/1194